### ISSUE TITLE
sh not found after restored cache

### ISSUE THREAD

**Issue by vanthao03596** [actions/cache#561]
After use actions/cache, my workflow run-tests not work. Before use actions/cache everything work fine

**here is output:**
```
yarn run v1.22.5
$ npm run production

> @ production /home/runner/work/jetstream-admin/jetstream-admin
> mix --production

sh: 1: mix: not found
npm ERR! code ELIFECYCLE
npm ERR! syscall spawn
npm ERR! file sh
npm ERR! errno ENOENT
npm ERR! @ production: `mix --production`
npm ERR! spawn ENOENT
npm ERR! 
npm ERR! Failed at the @ production script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/runner/.npm/_logs/2021-04-02T07_31_45_191Z-debug.log
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Here is my workflow :**
```
- name: Get yarn cache directory path
    id: yarn-cache-dir-path
    run: echo "::set-output name=dir::$(yarn cache dir)"

- uses: actions/cache@v2
  id: yarn-cache # use this to check for `cache-hit` (`steps.yarn-cache.outputs.cache-hit != 'true'`)
  with:
    path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
    key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
    restore-keys: |
      ${{ runner.os }}-yarn-
- name: Install Dependencies
  if: steps.yarn-cache.outputs.cache-hit != 'true'
  run: yarn install

- name: Build
  run: |
    npm config set scripts-prepend-node-path true
    yarn prod
```

**Comment by josantana** [comment]
Using the exact same workflow above, but for linting. Cache is restored successfully, but when I try to run any command, it looks like I got nothing installed.

![Screen Shot 2021-04-03 at 12 27 07](https://user-images.githubusercontent.com/5032844/113483087-12f58a00-9478-11eb-863a-177464abf3d4.png)

I've tried to run `yarn list` before the "Lint" line, which returned a list of all installed dependencies. `eslint` was listed as expected. I don't know why it is not executed.
